cli/src/commands/extensions/api.rs

```diff
@@ -465,123 +405,83 @@ fn run_sandboxed(op_state: Rc<RefCell<OpState>>, process: Process) -> Result<Pro
-            let mut code = None;
-            if libc::WIFEXITED(exit_code) {
-                code = Some(libc::WEXITSTATUS(exit_code));
-            } else if libc::WIFSIGNALED(exit_code) {
-                signal = Some(libc::WTERMSIG(exit_code));
-            } else if libc::WIFSTOPPED(exit_code) {
-                signal = Some(libc::WSTOPSIG(exit_code));
-            }
+    // Add sandbox subcommand argument.
+    let mut sandbox_args = Vec::with_capacity(args.len());
+    sandbox_args.push("sandbox".into());
 
-            // Read STDOUT/STDERR from pipe.
-            let mut stdout = String::new();
-            if let Some(mut stdout_fd) = stdout_fds.parent {
-                stdout_fd.read_to_string(&mut stdout)?;
-            }
-            let mut stderr = String::new();
-            if let Some(mut stderr_fd) = stderr_fds.parent {
-                stderr_fd.read_to_string(&mut stderr)?;
-            }
+    // Create CLI arguments for `phylum sandbox` permission exceptions.
+    add_permission_args(&mut sandbox_args, &resolved_permissions, strict)?;
 
-            Ok(ProcessOutput { stdout, stderr, success: code == Some(0), signal, code })
-        },
+    // Add sandboxed command arguments.
+    sandbox_args.push("--".into());
+    sandbox_args.push(cmd.into());
+    for arg in &args {
+        sandbox_args.push(arg.into());
     }
-}
 
-/// Lock down the current process.
-#[cfg(unix)]
-fn lock_process(exceptions: permissions::Permissions, strict: bool) -> Result<()> {
-    let home_dir = dirs::home_dir()?;
+    // Execute sandboxed command.
+    let output = Command::new(env::current_exe()?)
+        .args(sandbox_args.iter_mut().map(|arg| arg.to_mut()))
+        .stdin(stdin)
+        .stdout(stdout)
+        .stderr(stderr)
+        .output()?;
+
+    Ok(ProcessOutput {
+        stdout: String::from_utf8_lossy(&output.stdout).into_owned(),
+        stderr: String::from_utf8_lossy(&output.stderr).into_owned(),
+        success: output.status.success(),
+        signal: output.status.signal(),
+        code: output.status.code(),
+    })
+}
 
-    let mut birdcage = if strict { Birdcage::new()? } else { permissions::default_sandbox()? };
+/// Convert [permissions::Permissions] to arguments for `phylum sandbox`.
+fn add_permission_args<'a>(
+    sandbox_args: &mut Vec<Cow<'a, str>>,
+    permissions: &'a permissions::Permissions,
+    strict: bool,
+) -> Result<()> {
+    if strict {
+        sandbox_args.push("--strict".into());
+    }
 
-    // Apply filesystem exceptions.
-    for path in exceptions.read.sandbox_paths().iter() {
+    // Add filesystem exception arguments.
+    let home_dir = dirs::home_dir()?;
+    for path in permissions.read.sandbox_paths().iter() {
         let path = dirs::expand_home_path(path, &home_dir);
-        permissions::add_exception(&mut birdcage, Exception::Read(path))?;
+        sandbox_args.push("--allow-read".into());
+        sandbox_args.push(path.to_string_lossy().into_owned().into());
```

Comment:
`Cow -> String -> Cow`? Do the lifetimes not match? Should both be `'a` based on the function signature.